lib/consul/async/vault_endpoint.rb

```diff
@@ -209,25 +211,24 @@ def fetch
           inactivity_timeout: 1, # default connection inactivity (post-setup) timeout
         }
         connection = EventMachine::HttpRequest.new(conf.base_url, options)
-        cb = proc do |n|
-          http = connection.send(http_method.downcase, build_request()) # Under the hood: c.send('get', {stuff}) === c.get({stuff})
+        cb = proc do |_|
+          http = connection.send(http_method.downcase, build_request) # Under the hood: c.send('get', {stuff}) === c.get({stuff})
           http.callback do
+            http_result = HttpResponse.new(http.dup.freeze, default_value)
             if enforce_json_200 && http.response_header.status != 200
-              _handle_error(http) { connection = EventMachine::HttpRequest.new(conf.base_url, options) }
+              _handle_error(http_result) { connection = EventMachine::HttpRequest.new(conf.base_url, options) }
```

Comment:
I believe the HTTPResponse object we defined is perfectible and would like in a further review to just add json preformatting to the EventMachine callback object
It is mostly to avoid parsing multiple time data in case of "legit" errors from server like {"errors": ["operation not supported"]}
The http.error ( transport errors, mostly ) + status_code + server response is enought imo for this simple review

but i agree we want to have better error handling with more information passed in (maybe in debug mode only)

Reply:
One could argue that in any case we would parse it only once since we exit the callback at this point tho